src/components/atoms/SupplierCoupons/index.tsx

```diff
@@ -1,62 +1,116 @@
   return (
-    <div className="bg-primary-main h-auto pl-4 rounded-full flex items-center gap-3 cursor-pointer">
-      <h2 className=" font-bold text-white text-xl">{discount}%</h2>
+    <div
+      className={`${
+        status == "used"
+          ? "bg-secondary-main"
+          : status == "expired"
+          ? "bg-generic-alertRed"
+          : "bg-primary-main"
+      } h-auto pl-5 rounded-full flex items-center gap-2 cursor-pointer`}
+    >
+      <h2
+        className={`font-semibold text-xl ${
+          status == "used" ? "text-primary-main" : "text-white"
+        }`}
+      >
+        {discount}%
+      </h2>
 
       <div
         className={classNames(
-          "rounded-full bg-white w-full py-3 gap-4 -m-[1px] hover:shadow-md",
+          "rounded-full bg-white w-full py-3 gap-2 -m-[1px] hover:shadow-md",
         )}
       >
         <div className="Rounded-full flex items-center justify-evenly w-full">
           <Image
             className="h-10 w-auto"
             alt="Coupons Image"
-            src={CouponPrimary}
+            src={status == "expired" ? CouponRed : CouponPrimary}
           />
-
           <span className="bg-gray-300 w-0.5 h-12"></span>
-          <div className="flex flex-col gap-1.5 text-xs">
-            <p className="flex font-semibold items-center text-generic-alertGreen">
-              <Image
-                className="h-3.5 pr-2 w-auto"
-                alt="coupon-black"
-                src={ShoppingCartGreen}
-                color="white"
-              />
-              faltam {unintsAmount} unidades
-            </p>
-            <p className="flex font-semibold items-center text-generic-alertRed">
-              <Image
-                className="h-3.5 pr-2 w-auto"
-                alt="coupon-black"
-                src={ClockCircleRed}
-              />
-              termina em {expirateTime} dias
-            </p>
-          </div>
           <div>
-            <Image className="h-6 w-auto" alt="arrow right" src={icon} />
+            <p className="text-xs pb-2 font-semibold text-black">{title}</p>
+            {status == "active" && (
+              <div className="flex flex-col gap-1.5 text-xs">
+                <p className="flex font-semibold items-center text-generic-alertGreen">
+                  <Image
+                    className="h-3.5 pr-2 w-auto"
+                    alt="coupon-black"
+                    src={ShoppingCartGreen}
+                    color="white"
+                  />
+                  faltam {unintsAmount} unidades
+                </p>
+                <p className="flex font-semibold items-center text-generic-alertRed">
+                  <Image
+                    className="h-3.5 pr-2 w-auto"
+                    alt="coupon-black"
+                    src={ClockCircleRed}
+                  />
+                  termina em {expirateTime}
+                </p>
+              </div>
+            )}
+            {status == "used" && (
```

Comment:
```suggestion
            {status === "used" && (
```

Reply:
U can use a ObjectMap to make this. 

const STATUS_COMPONENTS = {
used: <div>...</div>,
expired: <div>...</div>,
...
}


And remove all if ternarys for it: `{STATUS_COMPONENTS[status] && STATUS_COMPONENTS[status]}`